server/controllers/complaintHandler.js

```diff
@@ -0,0 +1,118 @@
+const DocComplaint = require('../database/models/docComplaint');
+const UserComplaint = require('../database/models/userComplaint');
+const authentication = ('../authentication');
+
+const createDocComplaint = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.redirect('/');
+  }
+  else {
+    let doccomplaintinst = new DocComplaint({
+      text: req.body.text,
+      fromId: req.body.fromId,
+      revisionId: req.body.revisionId,
+      docId: req.body.docId 
+    });
+    
+    doccomplaintinst.save(function (err, doccomp) {
+      if (err) {
+        res.status(500).json({error:"Could not create the Document Complaint."});
+      }
+      else {
+        console.log("Document Complaint saved to DocComplaints collection.");
+        res.status(200).json({msg:"Document Complaint Created."});
+      }
+    });
+  }
+}
+
+const retrieveDocComplaint = (req, res) => {
+  DocComplaint.findOne({_id:req.params.complaintId}, function(err, result){
+    if (err) {
```

Comment:
Check correct error handling and response status [here](https://github.com/chc5/gitgud/blob/master/server/controllers/docHandlers.js#L30)

Reply:
ok